lib/bundlex/toolchain/common/unix.ex

```diff
@@ -36,6 +36,23 @@ defmodule Bundlex.Toolchain.Common.Unix do
       compile_commands ++ link_commands(native, link, output, objects, options, native_interface)
   end
 
+  defp resolve_compiler_flags(compiler_flags, native_interface) do
+    compiler_flags
+    |> add_interface_macro_flag(native_interface)
+    |> Enum.join(" ")
+  end
+
+  defp add_interface_macro_flag(compiler_flags, native_interface) do
+    macro_flag =
+      cond do
+        native_interface == :nif -> "-DNIF"
+        native_interface == :cnode -> "-DCNODE"
+        native_interface == :port -> "-DPORT"
```

Comment:
- this will fail when `native_interface` is nil instead of adding no flag
- flags for interfaces can be generated with `"-D#{native_interface |> Atom.to_string() |> String.upcase()}"`

Reply:
and prefix the defines with `BUDNLEX_`, like `BUNDLEX_CNODE`

Reply:
fixed in aa9dbd3